### ISSUE TITLE
Cop idea: lint for inconsistent safe navigation

### ISSUE THREAD

**Issue by vlad-pisanov** [rubocop/rubocop#13142]
Not entirely sure if this makes sense, but something I often see in our codebase is the same method/variable being used with both safe navigation and without. This suggests that either a) the dev forgot to include safe navigation in one expression, or b) the safe navigation isn't needed in either expression. e.g.

```ruby
# bad
puts user.address&.street # <- one of these
puts user.address.country # <- is wrong
```

obviously we can't autocorrect this, but it would be useful to point out

**Comment by bbatsov** [comment]
Closing — labeled good first issue but no takers in 1.5 years. If someone wants to implement a lint for inconsistent safe navigation, please reopen.
